File: ExampleProject/ExampleProject/ViewController.m
//
//  ViewController.m
//  ExampleProject
//
//  Created by Иван on 5/16/19.
//  Copyright © 2019 Ivan. All rights reserved.
//

#import "ViewController.h"
#import <ISKeystrokeDynamics/ISKeystrokeDynamics.h>

@interface ViewController () <UITextFieldDelegate>
@property (weak, nonatomic) IBOutlet ISTextField *textField;
@property (weak, nonatomic) IBOutlet UILabel *resultLabel;
@property (weak, nonatomic) IBOutlet UIButton *resetButton;

@property (nonatomic, strong) ISKeystrokeDynamicsRecognitionModulesHandler *kdrmh;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    _textField.delegate = self;
    _kdrmh = [[ISKeystrokeDynamicsRecognitionModulesHandler alloc] initWithTextField:_textField];
    _resetButton.layer.masksToBounds = YES;
    _resetButton.layer.cornerRadius = 20;
}

- (BOOL)textFieldShouldReturn:(UITextField *)textField {
    [self userDidEnterPassword];
    return YES;
}

- (void)userDidEnterPassword {
    [_textField resignFirstResponder];
    // Check password first!!!!!!!!!!
    if ([_textField.text isEqualToString:@"UsersPassword"]) {
        NSNumber *result = [_kdrmh startProcessingWithDataAmount:12 andMode:ISModeManhattanFiltered];
        if (result != nil) {
            if (fabs([result doubleValue]) <= 65) {
                [_resultLabel setText:@"Real user!"];
            } else if (fabs([result doubleValue]) > 65) {
                [_resultLabel setText:@"Fake user!"];
            }
        }
    } else {
        UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Error" message:@"Wrong password!" preferredStyle:UIAlertControllerStyleAlert];
        [alert addAction:[UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:nil]];
        [self presentViewController:alert animated:YES completion:nil];
        _textField.text = [NSString string];
    }
}
- (IBAction)resetFramewrokPressed:(id)sender {
    [_kdrmh resetData];
}

@end

File: ExampleProjectSwift/ExampleProjectSwift/ViewController.swift
//
//  ViewController.swift
//  ExampleProjectSwift
//
//  Created by Иван on 5/17/19.
//  Copyright © 2019 Ivan. All rights reserved.
//

import UIKit
import ISKeystrokeDynamics

class ViewController: UIViewController, UITextFieldDelegate {
    @IBOutlet weak var textField: ISTextField!
    @IBOutlet weak var resultLabel: UILabel!
    @IBOutlet weak var resetButton: UIButton!
    var kdrmh: ISKeystrokeDynamicsRecognitionModulesHandler!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        textField.delegate = self
        kdrmh = ISKeystrokeDynamicsRecognitionModulesHandler(textField: textField)
        resetButton.layer.masksToBounds = true
        resetButton.layer.cornerRadius = 20
    }
    
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        userDidEnterPassword()
        return true;
    }
    
    func userDidEnterPassword() {
        textField.resignFirstResponder()
        if (textField.text == "TestPassword") {
            let result = kdrmh.startProcessing(withDataAmount: 12, andMode: .manhattanFiltered)
            if let number = result {
                if fabs(number.doubleValue) <= 65 {
                    resultLabel.text = "Real user!"
                } else if fabs(number.doubleValue) > 65 {
                    resultLabel.text = "Fake user!"
                }
            }
        } else {
            let alert = UIAlertController(title: "Error", message: "Wrong password!", preferredStyle: .alert)
            alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
            present(alert, animated: true, completion: nil)
            textField.text = String();
        }
    }
    
    @IBAction func resetFrameworkPressed(_ sender: Any) {
        kdrmh.resetData()
    }
    
}

